test/test-agent-stopped.js

```diff
@@ -21,9 +21,14 @@ var http = require('http');
 
 describe('test-agent-stopped', function() {
   var agent;
-  before(function() {
+  before(function(done) {
     delete process.env.GCLOUD_PROJECT;
     agent = require('..').start();
+    // Wait 300ms for agent to fail getting remote project id.
```

Comment:
In the debug agent we expose an event emitter on the `Debuglet` to expose lifecycle events to the tests. The debuglet concept is a private concept, so it is easy to implement test-only interface on it.